### DISCUSSION TITLE
Improve error reporting on unpermissioned publish

### DISCUSSION THREAD

**Discussion by NiloCK** [npm/feedback#521] [CLI]
### Problem:

I recently attempted to `npm publish` a project owned by an organization that I wasn't a member of. Returned error was a `404`, indicating (incorrectly) that the target package does not exist.

``` shell
npm ERR! code E404
npm ERR! 404 Not Found - PUT https://registry.npmjs.org/@organization%2fpackage - Not found
npm ERR! 404 
npm ERR! 404  '@organization/package@0.0.5' is not in the npm registry.
npm ERR! 404 You should bug the author to publish it (or use the name yourself!)
npm ERR! 404 
npm ERR! 404 Note that you can also install from a
npm ERR! 404 tarball, folder, http url, or git url.
```

### Expected behaviour:

A `401` error code would:
- not be providing incorrect information (the XXX is not in the npm registry report above was not accurate)
- provide the user with actionable feedback ("I'm not logged in!", "I should request publishing rights to this package!")

**Comment by MylesBorins** [comment]
@NiloCK Was the package you were trying to publish a public package or a private package?

**Reply by NiloCK** [reply]
Public. @statechannels/exit-format if it adds anything to the discussion - I left it out in the name of speaking generically.

**Comment by targos** [comment]
Here's an example where a publish of a public package failed with this error: https://github.com/image-js/iobuffer/runs/3868923556?check_suite_focus=true

I had revoked the automation token used in this repo by mistake. Took me some time to understand what it was.
